Assets/Scripts/Movement.cs

```diff
@@ -81,9 +98,124 @@ private void FixedUpdate()
             Velocity.y = -2f;
         }
 
-     
+        findWhipPoint();
+    }
+
+    private void ChargingTorch(InputAction.CallbackContext obj)
+    {
+        
+        print("Holding");
+        isTorchButtonHeldDown = true;
+       
+    }
+
+    private void WhileChargingTorchThrow()
+    {
+        if (isTorchButtonHeldDown)
+        {
+            Vector3 CurrentTorchPosition = HeldTorch.transform.localPosition;
+            HeldTorch.transform.localPosition = new Vector3(CurrentTorchPosition.x, CurrentTorchPosition.y - 0.1f, CurrentTorchPosition.z);
+        } else
+        {
+            HeldTorch.transform.localPosition = DefaultTorchPosition;
+        }
+    }
+
+    private void TossTorch(InputAction.CallbackContext obj)
+    {
+            print(obj.duration);
+
+            //Direction Correction
+            Vector3 forceDirection = camera.transform.forward;
+            RaycastHit hit;
+
+            if(Physics.Raycast(camera.transform.position, forceDirection, out hit, 500f))
+            {
+            forceDirection = (hit.point - TorchSpawnPoint.transform.position).normalized;
+            }
+
+            ForwardsForce += ForwardsForce * (float)obj.duration * TorchForceMultiplier;
+            GameObject TorchInstance = Instantiate(Torch, TorchSpawnPoint.transform.position, camera.transform.rotation);
+            Rigidbody torch_rb = TorchInstance.GetComponent<Rigidbody>();
+            torch_rb.velocity = this.Velocity;
+            Vector3 ForceToAdd = forceDirection * ForwardsForce + TorchSpawnPoint.transform.up * UpwardsForce;
+            torch_rb.AddForce(ForceToAdd, ForceMode.Impulse);
+            torch_rb.AddTorque(transform.up * Random.Range(0, 20));
```

Comment:
I think this may be the direction you had in mind to make the torches tumble end over end while they are thrown, tested and this looks better to me:
```suggestion
            torch_rb.AddTorque(transform.right * Random.Range(0, 20));
```